### DISCUSSION TITLE
hub/datasets/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#12995] [Docs]
# hub/datasets/

Learn how Ultralytics HUB datasets streamline your ML workflow. Upload, format, validate, access, share, edit or delete datasets for Ultralytics YOLO model training.

https://docs.ultralytics.com/hub/datasets/

**Comment by ninopleno** [comment]
Hi there..... How should my YAML file looks like when uploading my dataset to your hub? I tried all possible combinations based on the instruction in `coco8.yaml` without success. Here is my dataset structure:

```shell
 |-- vehicle_v1/
    |-- images/
        |-- test/
        |-- train/
        |-- val/
    |-- labels/
        |-- test/
        |-- train/
        |-- val/
    |-- vehicle_v1.yaml
```

My YAML file looks like this:

```yaml
path:
train: [vehicle_v1/images/train, vehicle_v1/labels/train]
val: [vehicle_v1/images/val, vehicle_v1/labels/val]
test: [vehicle_v1/images/test, vehicle_v1/labels/test]

names:
  0: Armoured
  1: Civilian

download:

```

Any guidance and feedback are appreciated!

**Reply by barisyigit44** [reply]
Hello, you can download the example dataset. Update your file by looking at this .zip file.
Example dataset : https://github.com/ultralytics/hub/blob/main/example_datasets/coco8.zip,

You can also click the "upload dataset" button on the Ultralytics Hub homepage and look at the bottom of the window, you can see how the dataset should be.
I learned thanks to this example section, thank you to whoever made it.

**Reply by pderrenger** [reply]
Hello! It's great to hear that the example section was helpful for you. If you need to reference the structure directly from the example dataset, you can indeed download and unzip the COCO8 dataset from the provided link to see the exact format. Additionally, the "Upload Dataset" feature on the Ultralytics Hub homepage offers guidance on how to structure your dataset correctly. If you have any more questions or need further assistance, feel free to ask. Happy modeling! 🚀

**Comment by oren123456** [comment]
Hi
Is it possible to add more data to an existing dataset. 
For instance, add 100 more images/labels to the train directory?

**Reply by pderrenger** [reply]
@oren123456 hi there!

Yes, you can definitely add more data to an existing dataset on Ultralytics HUB. To add 100 more images/labels to your train directory, follow these steps:

1. **Prepare Your Additional Data**: Ensure your new images and labels follow the same structure and naming conventions as your existing dataset.

2. **Update the Dataset Directory**: Place the new images and labels in the appropriate subdirectories (e.g., `images/train` and `labels/train`).

3. **Update the Dataset YAML**: If necessary, update your dataset YAML file to reflect any changes, such as the number of images.

4. **Zip the Updated Dataset**: Zip the entire dataset directory again. For example:
    ```bash
    zip -r coco8.zip coco8
    ```

5. **Validate the Dataset**: Before uploading, validate the updated dataset to ensure it's correctly formatted:
    ```python
    from ultralytics.hub import check_dataset
    check_dataset("path/to/coco8.zip", task="detect")
    ```

6. **Upload the Updated Dataset**: Go to the Datasets page on Ultralytics HUB, click the **Upload Dataset** button, and upload your updated ZIP file.

For more detailed guidance, you can refer to the [Ultralytics HUB Datasets documentation](https://docs.ultralytics.com/hub/datasets/).

If you encounter any issues or need further assistance, feel free to ask. Happy training! 🚀

**Comment by Brenda-Caroline** [comment]
Hi, I made some adaptations to a dataset. Then delete the dataset that was in the hub to place the new dataset. But it always makes an error. Displays the message "Unable to create dataset"!

The dataset presents the recommended structure.

**Reply by glenn-jocher** [reply]
@Brenda-Caroline it seems there might be an issue with the dataset structure or format. Please ensure your dataset YAML, directory, and ZIP file have the same name and validate the dataset using `check_dataset("path/to/dataset.zip", task="detect")` before uploading. If the issue persists, double-check the dataset paths and file permissions.

**Reply by Brenda-Caroline** [reply]
Thank you Glenn, I got it.

Now I have some questions regarding the training, parameters and results of the model.

1) Is there a formula, a recommendation that helps define the number of epochs to train a given model? For example, a model that has 3 epochs is recommended to initially use x epochs.

2) How can I identify when the model is overfitting during training? Is it possible to see this in the loss graph when the line corresponding to the train is increasing?

3) Regarding the Cache and Device parameters, if I set them as default, what are the choices for execution? Is it RAM and GPU?
I did training using RAM and GPU and another training without RAM and GPU, the results were different. What are the influences of these metrics on the model? One of the loss graphs in the training without RAM and GPU did not make the loss curve as it should be, it was closer to a straight line close to 0. Why does this happen?


4) Still regarding the loss graphs, they mean that when training is done, the graph starts at the top and as the epochs go by, the graph gets smaller. Does this mean that as the training progresses, the model is learning and consequently making fewer mistakes in its predictions? And the smaller this graph is at the end of the training, the better?

**Reply by glenn-jocher** [reply]
1. There's no fixed formula for epochs; start with a baseline and adjust based on validation performance.
2. Overfitting is indicated by a decreasing training loss and increasing validation loss.
3. Default settings use GPU and RAM if available; differences in results may arise from hardware efficiency.
4. Yes, a decreasing loss graph typically indicates the model is learning effectively.

**Reply by femiadewoleadewole** [reply]
#- [ ]

**Reply by glenn-jocher** [reply]
@femiadewoleadewole if you're encountering "Unable to create dataset," please ensure your dataset ZIP file is correctly structured and validated using `check_dataset()`. If the issue persists, double-check the file paths and permissions.

**Reply by Brenda-Caroline** [reply]
So if I increase the validation loss values ​​a little, can I consider it overfitting? My model increased the value mainly in training, but the validation value also increased a little. Which epoch do I consider to have started overfitting? When the value first increased? And how can I solve this? By decreasing the patience, decreasing the number of epochs trained or changing the model?

I trained some models with yolov8x.yaml (it was not pre-trained) and usually downloads the file corresponding to the yolov8n.pt model (pre-trained). Is this correct?

**Reply by glenn-jocher** [reply]
@Brenda-Caroline to address overfitting, monitor when validation loss starts increasing while training loss decreases. Consider reducing epochs, adjusting patience, or trying a different model. For training, ensure you're using the correct pretrained weights. If you encounter further issues, please refer to the documentation for guidance.
